awx/ui/client/test/unit/layout.spec.js

```diff
@@ -0,0 +1,162 @@
+                expect(controller.isSuperUser).toBe(true);
+
+                $rootScope.current_user = 'two';
+                $rootScope.user_is_superuser = false;
+                $rootScope.user_is_system_auditor = true;
+                scope.$digest();
+                expect(controller.isSuperUser).toBe(true);
+
+                $rootScope.current_user = 'three';
+                $rootScope.user_is_superuser = true;
+                $rootScope.user_is_system_auditor = true;
+                scope.$digest();
+                expect(controller.isSuperUser).toBe(true);
+
+                $rootScope.current_user = 'four';
+                $rootScope.user_is_superuser = false;
+                $rootScope.user_is_system_auditor = false;
+                scope.$digest();
+                expect(controller.isSuperUser).toBe(false);
+            });
+
+            it('currentUsername', () => {
+                expect(controller.currentUsername).toBeTruthy();
+                expect(controller.currentUsername).toBe('admin');
+            });
+
+            it('currentUserId', () => {
+                expect(controller.currentUserId).toBeTruthy();
+                expect(controller.currentUserId).toBe(1);
+            });
+
+        });
+
+        describe('$root.socketStatus watcher should assign newStatus to', () => {
+            let statuses = ['connecting', 'error', 'ok'];
+
+            it('socketState', () => {
+                _.forEach(statuses, (status) => {
+                    $rootScope.socketStatus = status;
+                    scope.$digest();
+                    expect(controller.socketState).toBeTruthy();
+                    expect(controller.socketState).toBe(status);
+                });
+            });
+
+            it('socketIconClass', () => {
+                _.forEach(statuses, (status) => {
+                    $rootScope.socketStatus = status;
+                    scope.$digest();
+                    expect(controller.socketIconClass).toBe(`icon-socket-${status}`);
+                });
+            });
+        });
+
+        describe('$root.licenseMissing watcher should assign true or false to', () => {
+            it('licenseIsMissing', () => {
+                $rootScope.licenseMissing = true;
+                scope.$digest();
+                expect(controller.licenseIsMissing).toBe(true);
+
+                $rootScope.licenseMissing = false;
+                scope.$digest();
+                expect(controller.licenseIsMissing).toBe(false);
+            });
+        });
+
+        describe('getString()', () => {
+            it('calls ComponentsStrings get() method', angular.mock.inject((_ComponentsStrings_) => {
+                spyOn(_ComponentsStrings_, 'get');
+                controller.getString('VIEW_DOCS')
+                expect(_ComponentsStrings_.get).toHaveBeenCalled();
+            }));
+
+            it('ComponentsStrings get() method  should throw an error', () => {
```

Comment:
I think the name of this test case could use a little more context: "It should throw an error when no matching key is given"